src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java

```diff
@@ -0,0 +1,138 @@
+    * Returns false if AVX is not supported. <br>
+    * Returns false if the native library cannot be loaded for any reason. <br>
+    *
+            * @param tempDir  directory where the native library is extracted or null to use the system temp directory
+    * @return  true if the native library is supported and loaded, false otherwise
+    */
+
+    @Override
+    public synchronized boolean load(File tempDir) {
+        boolean isLoaded = gklUtils.load(null);
+
+        if(!isLoaded)
+        {
+            logger.warn("Intel GKL Utils not loaded");
+            return false;
+        }
+
+        if (!gklUtils.isAvxSupported()) {
+            return false;
+        }
+
+        if (!NativeLibraryLoader.load(tempDir, NATIVE_LIBRARY_NAME)) {
+            return false;
+        }
+        if (!initialized) {
+
+            initialized = true;
+        }
+        return true;
+    }
+
+    /**
+     * Initialize native SmithWaterman with the supplied args.
+     *
+     * @param args the args used to configure native SmithWaterman
+     */
+
+    public void initialize(SWAlignerArguments args) {
+        if(args == null)
+        {
+                args = new SWAlignerArguments(SWAlignerArguments.OverhangStrategy.SOFTCLIP,1,-1,1,1);
+        }
+        else {
+            w_extend = args.w_extend;
+            w_match = args.w_match;
+            w_mismatch = args.w_mismatch;
+            w_open = args.w_open;
+            strategy = args.strategy;
+        }
+
+    }
+
+    public int w_extend;
+    public int w_match;
+    public int w_mismatch;
+    public int w_open;
+    public SWAlignerArguments.OverhangStrategy strategy;
+    private native static int alignNative(byte[] refArray, byte[] altArray, byte[] cigar, int match, int mismatch, int open, int extend, int strategy);
+
+    /**
+     *Implements the native implementation of SmithWaterman, and returns the Cigar String and alignment_offset
+     *
+     * @param refArray array of reference data
+     * @param altArray array of alternate data
+     *
+     */
+
+    public SWAlignmentResult align(byte[] refArray, byte[] altArray)
+    {
+        int intStrategy =  getStrategy(strategy);
+        byte[] cigar = new byte[200];
+        int offset = alignNative(refArray, altArray, cigar, w_match, w_mismatch, w_open, w_extend, intStrategy);
+        return new SWAlignmentResult(new String(cigar),offset);
+    }
+
+    public static int getStrategy(SWAlignerArguments.OverhangStrategy strategy)
+    {
+        int intStrategy=0;
+
+        switch(strategy){
+            case SOFTCLIP: intStrategy = 9;
+                return intStrategy;
+            case INDEL: intStrategy = 10;
+                return intStrategy;
+            case LEADING_INDEL: intStrategy = 11;
+                return intStrategy;
+            case IGNORE: intStrategy = 12;
+                return intStrategy;
+        }
+
+        return intStrategy;
+
+    }
```

Comment:
This is correct, but I think it would be clearer to avoid having all those return statements. Instead, we can have `break`s in there, since at the end we're returning `intStrategy` anyway:
```java
switch(strategy){
    case SOFTCLIP: intStrategy = 9;
        break;
    case INDEL: intStrategy = 10;
         break;
    // other cases...
}

return intStrategy;
```
This is a stylistic comment, so obviously you can take it or leave it.